README.md

```diff
@@ -28,22 +151,151 @@ mtx.myTranspose();
        ]
 ```
 
+Array.prototype.myTranspose = function(callback){
+  let grid = [];
+  for(let i = 0; i < mtx[0].length; i++){
+    grid[i] = []
+
+    for(let j = 0; j< mtx.length;j++){
+      grid[i][j] = mtx[j][i]
+
+    }
+  }
+  return grid
+};
+
+let mtx = [
+            [1, 2],
+            [3, 4],
+            [5, 6]
+          ]
+
+//console.log(transpose(mtx))
+let results = mtx.myTranspose((el) => {
+  console.log(el)
+})
+console.log(results)
+
 7. Use `myMap` to build a new array where every element is incremented by 10.
-8. Use `myMap` to build a new array where every non-string element is converted to an empty string. 
+
+Array.prototype.myForEach = function (callback) {
+  for(let i = 0; i < this.length; i++){
+    callback(this[i]);
+  }
+
+};
+
+Array.prototype.myMap = function(callback) {
+let newArray = [];
+this.myForEach((el) => {
+  newArray.push(callback(el))
+})
+return newArray
+};
+
+
+/*let arr2 = [1,2,3,4,5]
+let finalArray = arr2.myMap((el) => {
+  return el * 2
+
+});*/
+
+let arr = [1,2,3,4,5,6];
+
+let results = arr.myMap((el) => {
+  console.log(el * 10)
+});
+
+
+console.log(results)
+
+8. Use `myMap` to build a new array where every non-string element is converted to an empty string.
+Array.prototype.myForEach = function (callback) {
+  for(let i = 0; i < this.length; i++){
+    callback(this[i]);
+  }
+
+};
+
+Array.prototype.myMap = function(callback) {
+let newArray = [];
+this.myForEach((el) => {
+  newArray.push(callback(el))
+})
+return newArray
+};
+
+let arr = ["hello","world",1,2,"how",4]
+
+let results = arr.myMap((el)=> {
+if(typeof el === "number"){
```

Comment:
What if there are booleans in the array? What if there are other arrays? We want to convert those to empty strings too!